authentication/views.py

```diff
@@ -333,6 +333,33 @@ def delete(self, request):
         return Response("Company Deleted Successfully", status.HTTP_200_OK)
 
 
+class ClientListView(generics.GenericAPIView):
+    """
+    Return a list of registered clients
+    """
+    serializer_class = ClientSerializer
+    renderer_classes = (ClientJSONRenderer,)
+    permission_classes = (IsAuthenticated,)
+
+    def get_queryset(self):
+        return Client.active_objects.all()
+
+    def get(self, request):
+        serializer = self.serializer_class(self.get_queryset(), many=True)
+        if len(serializer.data) == 0:
```

Comment:
Good work on the implementation @Rhytah 

This doesn't really affect how the code works but just as a side note - empty lists in Python are considered `False`  so it would be more pythonic here to do:
```python
if not serializer.data:
```

This is recommended by [PEP 8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations):

> For sequences, (strings, lists, tuples), use the fact that empty sequences are false.
> 
```python
Yes: if not seq:
     if seq:
 
No:  if len(seq):
     if not len(seq):
```

You can try it out... see if it works.

Reply:
Hey @wangonya , does the `if not` give the impression that the list is a `boolean`

Reply:
@wangonya I have tried it out and works just fine. We learn something new every day.

Reply:
Not really @Tevinthuku. We're not checking that the list is implicitly `False` but rather `falsy`. Empty lists are falsy.

Reply:
Alright makes sense. 👍